lib/src/settings.rs

```diff
@@ -357,27 +357,22 @@ impl<'de> serde::Deserialize<'de> for HumanByteSize {
                 write!(formatter, "a size in bytes with an optional binary unit")
             }
 
-            fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
-            where
-                E: Error,
-            {
-                Ok(HumanByteSize(v))
-            }
-
             fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
             where
                 E: Error,
             {
-                let bytes = parse_human_byte_size(v).map_err(Error::custom)?;
-                Ok(HumanByteSize(bytes))
+                let res = v.parse::<u64>();
+                match res {
+                    Ok(bytes) => Ok(HumanByteSize(bytes)),
+                    Err(_) => {
+                        let bytes = parse_human_byte_size(v).map_err(Error::custom)?;
+                        Ok(HumanByteSize(bytes))
+                    }
+                }
             }
         }
 
-        if deserializer.is_human_readable() {
-            deserializer.deserialize_any(Visitor)
-        } else {
-            deserializer.deserialize_u64(Visitor)
-        }
+        deserializer.deserialize_str(Visitor)
```

Comment:
Maybe we can simplify it further by using `FromStr` or `TryFrom<String>`?

https://github.com/serde-rs/serde/issues/908
https://github.com/serde-rs/serde/issues/939

I think conversing through string works because config-rs is lax about scalar types.

Reply:
Thanks, this is much nicer!